### PR TITLE
Add move ability to attachments [SCI-9088]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-9088](https://scinote.atlassian.net/browse/SCI-9088)

### What was done
Add move ability to attachments

[SCI-9088]: https://scinote.atlassian.net/browse/SCI-9088?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/vue/shared/content/attachments.vue [hunk 1]
```diff
@@ -142,7 +144,7 @@
         return `${attachment.attributes.view_mode}Attachment`
       },
       deleteAttachment(id) {
-        this.$emit('attachment:deleted', id)
+        this.$emit('attachment:deleted', id, )
```
Comment (artoscinote) [inline app/javascript/vue/shared/content/attachments.vue:147]: Remove this comma and space.

**Review by artoscinote — CHANGES_REQUESTED** [review]
Fix linter issues.

**Review by artoscinote — APPROVED** [review]

File: app/controllers/assets_controller.rb [hunk 2]
```diff
@@ -65,6 +65,36 @@ def load_asset
...
+        object_to_update = @asset.step_asset
+        object_to_update.update!(step: target)
+        render json: { status: :ok }
+      elsif @assoc.is_a?(Result)
+        target = @assoc.my_module.results.find_by(id: params[:target_id])
+        object_to_update = @asset.result_asset
+        object_to_update.update!(result: target)
+        render json: { status: :ok }
+      end
+    rescue ActiveRecord::RecordInvalid
```
Comment (okriuchykhin) [inline app/controllers/assets_controller.rb:93]: will the transaction be rolled backed?

File: app/controllers/assets_controller.rb [hunk 3]
```diff
@@ -65,6 +65,36 @@ def load_asset
...
+    end
+  end
+
+  def move
+    ActiveRecord::Base.transaction do
+      if @assoc.is_a?(Step)
+        target = @assoc.protocol.steps.find_by(id: params[:target_id])
+        object_to_update = @asset.step_asset
+        object_to_update.update!(step: target)
+        render json: { status: :ok }
```
Comment (okriuchykhin) [inline app/controllers/assets_controller.rb:86]: why not simply use HTTP codes?

**Review by okriuchykhin — CHANGES_REQUESTED** [review]
check comments

**Review by okriuchykhin — APPROVED** [review]

**Pull request merged**